ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh

```diff
@@ -0,0 +1,363 @@
+          source "${SHARED_DIR}/proxy-conf.sh"
+      fi
+
+      export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+      mirror_support_tools
+  fi
+}
+
+wait_for_mcp() {
+  timeout=${1}
+  # Wait until MCO starts applying new machine config to nodes
+  log "Waiting for all MachineConfigPools to start updating..."
+  time oc wait mcp --all --for='condition=UPDATING=True' --timeout=300s &>/dev/null
+
+  log "Waiting for all MachineConfigPools to finish updating..."
+  time timeout "${timeout}" bash <<EOT
+    until
+      oc wait mcp --all --for='condition=UPDATED=True' --timeout=10s 2>/dev/null && \
+      oc wait mcp --all --for='condition=UPDATING=False' --timeout=10s 2>/dev/null && \
+      oc wait mcp --all --for='condition=DEGRADED=False' --timeout=10s;
+    do
+      sleep 10
+    done
+EOT
+}
+
+wait_for_co() {
+  timeout=${1}
+  log "Waiting for all ClusterOperators to update..."
+  time timeout "${timeout}" bash <<EOT
+  until
+    oc wait co --all --for='condition=AVAILABLE=True' --timeout=10s &>/dev/null && \
+    oc wait co --all --for='condition=PROGRESSING=False' --timeout=10s &>/dev/null && \
+    oc wait co --all --for='condition=DEGRADED=False' --timeout=10s &>/dev/null;
+  do
+    sleep 10
+  done
+EOT
+}
+
+wait_for_cno_cleanup() {
+  timeout=${1}
+  log "Waiting for CNO to clean migration status..."
+  time timeout "${timeout}" bash <<EOT
+  until
+    ! oc get network -o yaml | grep migration > /dev/null
+  do
+    echo "migration field is not cleaned by CNO"
+    sleep 10
+  done
+EOT
+}
+
+wait_for_final_mc() {
+  timeout=${1}
+  log "Waiting for final MachineConfigs..."
+  script=$(sed -e "s,%%data%%,${encoded_set_mtu},g" <<'EOT'
+  while [ "${is_final:-0}" = "0" ]; do
+    is_final=1
+    for pool in worker master; do
+      mc_name=$(oc get mc -o custom-columns=:metadata.name --sort-by=.metadata.creationTimestamp | grep rendered-$pool | tail -1)
+      mc=$(oc get mc $mc_name -o yaml || true)
+      [ -z "$mc" ] && is_final=0 && break
+      echo $mc | grep -q "mtu-migration.sh" && is_final=0 && break
+      echo $mc | grep -q "%%data%%" || (is_final=0 && break)
+    done
+    [ "$is_final" = "1" ] || sleep 10
+  done
+EOT
+)
+  time timeout "${timeout}" bash <<SCRIPT
+  $script
+SCRIPT
+}
+
+check_no_api_alerts() {
+  timeout=${1}
+  log "Checking for no API alerts during ${timeout} ..."
+  time timeout "${timeout}" bash <<'EOT' || [ "$?" = "124" ]
+  ALERT_HOSTNAME=$(oc get routes/prometheus-k8s -n openshift-monitoring -o json | jq -r '.spec.host')
+  ALERT_URL="https://${ALERT_HOSTNAME}/api/v1/alerts"
+	ALERT_TOKEN=$(oc -n openshift-monitoring sa get-token prometheus-k8s)
+  ALERTS="(KubeAggregatedAPIErrors|KubeAggregatedAPIDown)"
+  while true
+  do
+    alerts=$(curl --silent --insecure --header "Authorization: Bearer ${ALERT_TOKEN}" "${ALERT_URL}" | jq -r '.data.alerts[] | select(.state=="firing") | .labels.alertname // empty' | sort -u | tr '\n' ' ')
+    echo "alerts in firing state: ${alerts}"
```

Comment:
nit: echo something here indicating we are going to bail on the job because we these alerts.

Reply:
Ok on this one here.

On the one below we are already doing `echo "Failed to get default interface"`. Is it something else?

Reply:
I'm just used to digging through CI logs and sometimes it's not clear where a "FAIL" thing is just printed as
a log, or when that failure actually led to the exit 1 which is why the job is marked as a failure. so any verbosity
around that can help whoever is debugging later.

Reply:
got it, I will qualify all errors with `Error`.